ext/strings_test.go

```diff
@@ -0,0 +1,269 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ext
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/google/cel-go/cel"
+)
+
+// TODO: move these tests to a conformance test.
+var stringTests = []struct {
+	expr      string
+	err       string
+	parseOnly bool
+}{
+	// CharAt test.
+	{expr: `'tacocat'.charAt(3) == 'o'`},
+	// Indext of search string tests.
+	{expr: `'tacocat'.indexOf('a') == 1`},
+	{expr: `'tacocat'.indexOf('a', 3) == 5`},
+	{expr: `'tacocat'.indexOf('none') == -1`},
+	{expr: `'tacocat'.lastIndexOf('a') == 5`},
+	{expr: `'tacocat'.lastIndexOf('a', 3) == 1`},
+	{expr: `'tacocat'.lastIndexOf('none') == -1`},
+	// Replace tests
+	{expr: `"12 days 12 hours".replace("{0}", "2") == "12 days 12 hours"`},
+	{expr: `"{0} days {0} hours".replace("{0}", "2") == "2 days 2 hours"`},
+	{expr: `"{0} days {0} hours".replace("{0}", "2", 1).replace("{0}", "23") == "2 days 23 hours"`},
+	// Split tests.
+	{expr: `"hello world".split(" ") == ["hello", "world"]`},
+	{expr: `"hello world events!".split(" ", 0) == []`},
+	{expr: `"hello world events!".split(" ", 1) == ["hello world events!"]`},
+	{expr: `"hello world events!".split(" ", 2) == ["hello", "world events!"]`},
+	// Substring tests.
+	{expr: `"tacocat".substring(4) == "cat"`},
+	{expr: `"tacocat".substring(0, 4) == "taco"`},
+	{expr: `"tacocat".substring(4, 4) == ""`},
+	// Trim tests using the unicode standard for whitespace.
+	{expr: `" \f\n\r\t\vtext  ".trim() == "text"`},
+	{expr: `"\u0085\u00a0\u1680text".trim() == "text"`},
+	{expr: `"text\u2000\u2001\u2002\u2003\u2004\u2004\u2006\u2007\u2008\u2009".trim() == "text"`},
+	{expr: `"\u200atext\u2028\u2029\u202F\u205F\u3000".trim() == "text"`},
+	// Trim test with whitespace-like characters not included.
+	{expr: `"\u180etext\u200b\u200c\u200d\u2060\ufeff".trim()
+				== "\u180etext\u200b\u200c\u200d\u2060\ufeff"`},
+	// Error test cases based on checked expression usage.
```

Comment:
Still looking for unicode-containing tests for charAt, indexOf, lastIndexOf, replace, split, and substring.  These can be just single non-edge-case tests.

Reply:
I misunderstood, thought you were looking for any tests involving unicode.